src/core/fem/src/condition/4C_fem_condition_multipointcoupling.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef FOUR_C_FEM_CONDITION_MULTIPOINTCOUPLING_HPP
+#define FOUR_C_FEM_CONDITION_MULTIPOINTCOUPLING_HPP
+
+#include "4C_config.hpp"
+
+#include "4C_fem_condition.hpp"
+#include "4C_linalg_map.hpp"
+
+#include <Teuchos_Time.hpp>
+#include <Teuchos_TimeMonitor.hpp>
+
+#include <memory>
+#include <set>
+
+FOUR_C_NAMESPACE_OPEN
+
+namespace Core::FE
+{
+  class Discretization;
+}  // namespace Core::FE
+
+namespace Core::Conditions
+{
+  /*!
+  \brief  update node row map for point coupling conditions
+  master and connected slave nodes must be owned by the same processor
+  */
+  class MultipointcouplingConditions
+  {
+   public:
+    //! constructor which checks the discretization for point coupling conditions
+    MultipointcouplingConditions(
+        std::shared_ptr<Core::FE::Discretization> dis, bool verbose = true);
+
+    //! Destructor
+    virtual ~MultipointcouplingConditions() = default;
+
+    //! are there pointcoupling conditions in use
+    bool has_pcc() const { return has_pointcoupling_conditions_; }
+
+    //! make sure that proper node distribution is achieved for point coupling conditions
+    void redistribute_for_pointcoupling_conditions();
```

Comment:
This is called only once, directly after the constructor. So it can at least be private, but I would prefer to remove this completely form the header. If you want an implementation heleper, that is fine, but I would put it in the cpp file.